src/toil/test/src/toilContextManagerTest.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (C) 2015 UCSC Computational Genomics Lab
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from toil.common import Toil
+from toil.job import Job
+from toil.test import ToilTest
+
+
+class ToilContextManagerTest(ToilTest):
+    def testContextManger(self):
+        options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+        options.logLevel = "INFO"
+        with Toil(options) as toil:
+            toil.run(HelloWorld())
+
+    def testNoContextManger(self):
+        options = Job.Runner.getDefaultOptions(self._getTestJobStorePath())
+        options.logLevel = "INFO"
+        toil = Toil(options)
+        try:
+            toil.run(HelloWorld())
+            self.fail()
+        except RuntimeError:
```

Comment:
Use specific exception ToilContextManagerMisueException and use assertFails instead of try, except and fail.